src/Features/Core/Portable/CodeFixes/AddExplicitCast/AbstractAddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,399 @@
+        /// <param name="potentialConversionTypes"> Output (target expression, potential conversion type) pairs</param>
+        /// <returns>
+        /// True, if there is at least one potential conversion pair, and they are assigned to "potentialConversionTypes"
+        /// False, if there is no potential conversion pair.
+        /// </returns>
+        protected abstract bool TryGetTargetTypeInfo(Document document, SemanticModel semanticModel, SyntaxNode root,
+            string diagnosticId, TExpressionSyntax spanNode, CancellationToken cancellationToken,
+            out ImmutableArray<Tuple<TExpressionSyntax, ITypeSymbol>> potentialConversionTypes);
+
+        protected abstract bool IsObjectCreationExpression(TExpressionSyntax targetNode);
+
+        protected abstract bool IsConversionUserDefined(
+            SemanticModel semanticModel, TExpressionSyntax expression, ITypeSymbol type);
+
+        protected ImmutableArray<Tuple<TExpressionSyntax, ITypeSymbol>> FilterValidPotentialConversionTypes(
+            SemanticModel semanticModel,
+            ArrayBuilder<Tuple<TExpressionSyntax, ITypeSymbol>> mutablePotentialConversionTypes)
+        {
+            using var _ = ArrayBuilder<Tuple<TExpressionSyntax, ITypeSymbol>>.GetInstance(out var validPotentialConversionTypes);
+            foreach (var conversionTuple in mutablePotentialConversionTypes)
+            {
+                var targetNode = conversionTuple.Item1;
+                var targetNodeConversionType = conversionTuple.Item2;
+
+                // For cases like object creation expression. for example:
+                // Derived d = [||]new Base();
+                // It is always invalid except the target node has explicit conversion operator or is numeric.
+                if (IsObjectCreationExpression(targetNode)
+                    && !IsConversionUserDefined(semanticModel, targetNode, targetNodeConversionType))
+                {
+                    continue;
+                }
+
+                validPotentialConversionTypes.Add(conversionTuple);
+            }
+            return validPotentialConversionTypes.Distinct().ToImmutableArray();
+        }
+
+        protected abstract bool ClassifyConversionExists(
+            SemanticModel semanticModel, TExpressionSyntax expression, ITypeSymbol type);
+
+        protected abstract bool IsConversionIdentity(
+            SemanticModel semanticModel, TExpressionSyntax expression, ITypeSymbol type);
+
+        protected abstract SeparatedSyntaxList<SyntaxNode> GetArguments(SyntaxNode argumentList);
+
+        protected abstract SyntaxNode GenerateNewArgument(SyntaxNode oldArgument, ITypeSymbol conversionType);
+
+        protected abstract TExpressionSyntax? GetArgumentExpression(SyntaxNode argument);
+        protected abstract bool IsDeclarationExpression(TExpressionSyntax expression);
+
+        protected abstract string? TryGetName(SyntaxNode argument);
```

Comment:
is this getting the name of a named arg?  i.e. `foo: bar` => "foo"`?

Reply:
yeah this is getting the name of a named argument